### ISSUE TITLE
Random crash when going up (monster list subwindow)

### ISSUE THREAD

**Issue by angband-import** [angband/angband#3863]
**Reported by draconisPW on 14 Nov 2013 08:27 UTC**
This is very hard to reproduce. Logging the error message, I got something like "access violation at 0x459390". This seems to happen on line 250 in mon-util.c (display_monlist) which is:

```
v = &list[m_ptr->race->ridx];
```

Looking at the code, this is the only place in a "for (i = 1; i < cave_monster_max(cave); i++)" loop that there's no check for m_ptr->race ("skip dead monsters"). I presume this is the cause of the crash: m_ptr->race is null. Should be easy to fix.

**Comment by angband-import** [comment]
**Comment by bensemmler on 27 Nov 2013 03:47 UTC**
This is no longer relevant due to the new monster list code (commit f23c0f777f3d).

**Comment by angband-import** [comment]
**Comment by draconisPW on 27 Nov 2013 08:42 UTC**
Well, the crash won't happen, but instead you'll get an empty entry (NULL race) with all other info set... which will be used for the next live monster. So you'll get count incremented and so on. This happens in really rare circumstances and nobody would probably notice the off-by-one monster count, but I guess adding a check wouldn't hurt...

**Comment by angband-import** [comment]
**Modified by bensemmler on 27 Nov 2013 14:25 UTC**

**Comment by angband-import** [comment]
**Comment by takkaria on 29 Nov 2013 00:17 UTC**
Moving to 3.5 because open bugs.

**Comment by angband-import** [comment]
**Comment by takkaria on 1 Dec 2013 21:54 UTC**
Fixed in https://github.com/angband/angband/pull/302

**Comment by angband-import** [comment]
**Comment by draconisPW on 3 Dec 2013 15:43 UTC**
The fix incorrectly wipes the "count" and "asleep" arrays using WIPE, which works with structures. The best way is probably to wipe the whole entry here:

```
if (list->entries[j].race == NULL)
{
	/* We found an empty slot, so add this race here. */
	entry = &list->entries[j];
	WIPE(entry, monster_list_entry_t);
	entry->race = monster->race;
	break;
}
```

**Comment by angband-import** [comment]
**Modified by takkaria on 7 Dec 2013 17:06 UTC**

**Comment by angband-import** [comment]
**Comment by takkaria on 21 Dec 2013 14:40 UTC**
Fixed, finally, in 72e7da2
